usecases/objects/references_add.go

```diff
@@ -40,7 +40,10 @@ func (m *Manager) AddObjectReference(ctx context.Context, principal *models.Prin
 
 	ctx = classcache.ContextWithClassCache(ctx)
 
-	if err := m.authorizer.Authorize(principal, authorization.UPDATE, authorization.ShardsMetadata(input.Class, tenant)...); err != nil {
+	if err := m.authorizer.Authorize(principal, authorization.UPDATE, authorization.ShardsData(input.Class, tenant)...); err != nil {
+		return &Error{err.Error(), StatusForbidden, err}
+	}
+	if err := m.authorizer.Authorize(principal, authorization.READ, authorization.ShardsMetadata(input.Class, tenant)...); err != nil {
```

Comment:
There is a helper method `authorization.Shards` that merges these two resources together into a one-liner. Can we use that here instead (and elsewhere in the PR)?

Reply:
they are different verbs (R vs U)